### ISSUE TITLE
Video Doorbell Profile not working

### ISSUE THREAD

**Issue by mgoeller** [lukasroegner/homebridge-nello#13]
After I couldn't get the newest version of the plugin running I completely started from scratch. E.g. set up a fresh Raspberry Pi with camera module with the newest Raspbian, installed NodeJS, HomeBridge and the nello plugin. My config.json file looks as follows:

`{
    "bridge": {
        "name": "DebugHB",
        "username": "CC:22:3D:E3:CE:12",
        "port": 51825,
        "pin": "123-34-456"
    },
    
    "description": "Secondary HomeBridge for testing",

    "accessories": [
     ],

    "platforms": [
         {
            "platform" : "NelloPlatform",
            "name" : "NewNello",
            "username": "username",
            "password": "password",
            "lockTimeout": 5000,
            "locationUpdateInterval": 3600000,
            "exposeReachability": true,
            "doorbell": false,
            "videoDoorbell": true,
            "video": {
                "stream": "-re -i rtsp://184.72.239.149/vod/mp4:BigBuckBunny_175k.mov",
                "snapshotImage": "http://via.placeholder.com/1280x720",
                "maxWidth": 1280,
                "maxHeight": 720,
                "maxFPS": 30
            },
            "homekitUser": "FirstName LastName"
        }
     ]
}
`

I'm running homebridge as root; and tried the following:

1. doorbell: false, videoDorbell: true. Untouched "video" section, e.g. stream: "-re -i <your-url>". After pressing Add Accessory in the Home App I can see a camera with the same name as my lock, trying to add it with the correct 123-34-456 PIN keeps going forever and finally fails saying "Couldn't add accessory".
2. doorbell: false, videoDorbell: true, stream: see above, using the test rtsp stream. Same behaviour: can't add the extra camera accessory
3. doorbell: true, videoDorbell: false. I don't see the camera accessory anymore when wanting to add a New Accessory. But I dont get push notifications of someone ringing the Bell either. Is that the normal behaviour? I guess not. The other push notifications of a locked / unlocked Nello do work fine.
4. doorbell: true, videoDorbell: true. Can't see the camera accessory anymore after restarting the Raspberry Pi, restarting HomeBridge, removing the /persist and /accessory directory, removing the Homebridge instance from the Home App, changing the MAC-address-styled username in the config.json file.

The fourth state persists so far, no matter what I am doing. Any idea what could be the error?

**Comment by carlsendk** [comment]
Do you have an extra camera setup to your intercom? (no reference to the problem just me who wanted to know your setup)

**Comment by mgoeller** [comment]
Right now I am using the official Raspberry Pi camera to record the screen from my intercom (Rene Koch). Thats just a temporary solution until my EasyCap USB video grabber arrives (~10 $) to grab the analog video signal from the intercom directly.

**Comment by AlexanderBabel** [comment]
Hey there,

I updated the repo and added better support for the raspberry pi camera module v2.1

I brought one of these and tried it out. In my case it is working pretty well.

Just take a look in the develop branch and give it a shot.

Source: https://github.com/lukasroegner/homebridge-nello/tree/develop
npm cmd:
``` bash
npm i https://github.com/lukasroegner/homebridge-nello.git#develop
```

**Comment by AlexanderBabel** [comment]
Don't forget to adjust your config for the raspberry pi camera module. (https://github.com/lukasroegner/homebridge-nello/tree/develop)

**Comment by mgoeller** [comment]
Thank you Alex. I uninstalled the hello package and installed your dev branch on two different Raspberry PIs with cameras and each a HomeBridge instance. The nello door lock accessory is available and working, if I manually want to add the camera neither with the code nor directly does it show up. Not sure if this worked before but someone ringing the bell is logged in the console of my homebridge instance, see:

`Jun 28 15:17:19 raspberrypi homebridge[1198]: [6/28/2018, 3:17:19 PM] [nello.io] Someone rang the bell of the door with ID fa4ba81e-770d-4faa-a8fb-8b7677506074.
Jun 28 15:17:41 raspberrypi homebridge[1198]: [6/28/2018, 3:17:41 PM] [nello.io] Opening door at location with ID fa4ba81e-770d-4faa-a8fb-8b7677506074.
Jun 28 15:17:41 raspberrypi homebridge[1198]: [6/28/2018, 3:17:41 PM] [nello.io] Opened door at location with ID fa4ba81e-770d-4faa-a8fb-8b7677506074.
Jun 28 15:17:48 raspberrypi homebridge[1198]: [6/28/2018, 3:17:48 PM] [nello.io] Malte Göller opened the door with ID fa4ba81e-770d-4faa-a8fb-8b7677506074`

The following questions come to my mind:
1.  Can a HomeKit lock accessory even issue push notifications "Person A rang the bell", or only "Lock XY was un-/locked"?
2. I have a Logitech Circle 2 HomeKit camera which "simulates" two accessories when added to Home: one motion sensor, and one camera. The motion sensor is triggered and issues a push notification. In the rich notification a snapshot of the camera is shown. And if a lock accessory is in the same room, the unlock button as well. Shouldn't your nello plugin simulate such a motion sensor as well, to issue the "someone rang the bell" notification?
3. Is the line "videoDoorbell": false correct for the Raspberry Pi in your example config.json? Tried both options, with false and true and still cant add the camera. When I install the homebridge-camera-rpi plugin it shows up immediately.

Any idea?

**Comment by AlexanderBabel** [comment]
Hey there,

Your first question: A lock accessory can only be locked or unlocked nothing else.
Your second question: Nope HomeKit supports video doorbell accessories therefore the camera accessory you add to the Home App can you send a ring notification.
Your third question: you must set videoDoorbell and raspberrryPiCamera to true in order to see the accessory and add it to the Home App.

Heres my configuration:

``` json
        {
            "platform": "NelloPlatform",
            "name": "nello.io",
            "username": "USER",
            "password": "PASSWORD",
            "lockTimeout": 5000,
            "locationUpdateInterval": 604800000,
            "exposeReachability": true,
            "videoDoorbell": true,
            "raspberryPiCamera": true,
            "homekitUser": "Home Kit"
        }
```

I hope I could help you.

Greetings
Alex

**Comment by mgoeller** [comment]
Thank you very much Alex, I completely started from 0 again and suddenly everything works. One main problem was to add the module bcm2835-v4l2 to /etc/modules. Somehow the "door bell rang" notifications are buggy; working now but didn't yesterday for example. Could this be a problem of your webhook relay service?

Also in your example configuration for the raspberry pi, "videoDoorbell: false" is set to false. Not sure if this is intentionally but that was what related to my question 3 in the previous message.

Thank you very much!

**Comment by mgoeller** [comment]
Using the Raspberry Pi Camera works now perfectly. I've set up another RaspberryPi with a second Homebridge instance and the homebridge-nello plugin. I somehow think the camera integration has errors, since using the "original" homebridge-camera-ffmpeg plugin works fine. I proceed as follows:

1. install homebridge and homebridge-nello
2. I manually start homebridge from the terminal as root using the following config.json

>          {
> 			"platform" : "NelloPlatform",
> 			"name" : "nello.io",
> 			"username": "USER",
> 			"password": "PASSWD",
> 			"lockTimeout": 5000,
> 			"locationUpdateInterval": 3600000,
> 			"exposeReachability": true,
> 			"videoDoorbell": true,
> 			"video": {
> 				"stream": "-re -f video4linux2 -s 640x480 -i /dev/video0",
> 				"snapshotImage": "-re -f video4linux2 -ss 0.9 -i /dev/video0 -vframes 1",
> 				"maxWidth": 640,
> 				"maxHeight": 480,
> 				"maxFPS": 30
> 			}
> 		}

3. I add the Homebridge instance in the Home App scanning the QR code, it asks me automatically to add the lock as well.
4. I try to manually add the camera as an accessory. It usually shows up immediately after pressing "Dont Have a Code or Can't Scan" using the homebridge-camera-ffmpeg or homebridge-camera-rpi plugin.

I am trying since days to find the bug that it doesn't for the nello plugin, deleting the persist and accessories folders makes no difference. As do restarting all the iPhone or Raspberry. Creating a new home doesn't either, nor does deleting the Bridge and re-adding it to the Home. Do you have any suggestions?

**Comment by AlexanderBabel** [comment]
Hey there,

I updated the plugin to version 0.3.5 and added the vcodec option. This will eventually fix your problem.

Anyway your config looks like you are using a camera module. Try to use the raspberryPiCamera option.

Greetings
Alex

**Comment by mgoeller** [comment]
Thank you for your effort Alex. However I think we are misunderstanding each other. I have two Raspberry Pi's. Number one is working now fine using the Raspberry Pi Camera module. Number two should use "-re -f video4linux2 -s 640x480 -i /dev/video0" as a video source for the stream. Using 
this source with the official plugin from Khaos works fine.

The problem I constantly run into is that the Home App somehow cannot seem to find the camera accessory. E.g. after adding the HomeBridge instance and the automatically in the same process the "pure" door lock accessory I want to manually add the camera accessory, and get this view:
[Screenshot](https://i.imgur.com/FeIg7fP.png)
Using my code "123-34-456" doesn't help, Home cant find the camera. When using the official Khaos accessory the camera is immediately shown as "unsupported accessory" and can be directly added.

**Comment by mgoeller** [comment]
[This](https://i.imgur.com/jFLCvoJ.png) is what is shown when using the homebridge-camera-ffmpeg plugin. How can I help you to debug?

**Comment by AlexanderBabel** [comment]
Hey there,

a quick update:

Currently Im not able to process this issue because I have many stuff to learn for my exams the next weeks. I hope I will have some time during this period but if not I'll be the at the end of the month and go through this issue.

**Comment by DJay-X** [comment]
@AlexanderBabel Thanks Alexander. And fingers crossed for your exams. 🤞🏻 Very looking forward hearing from you here again.

**Comment by AlexanderBabel** [comment]
Hey there,

I think I found the issue. Could it be possible you are trying to add the camera accessory for the same Nello one? If the answer is true, the issue is, that both camera accessories have the same name and you already added one of them. Therefore the other one won't be displayed during the setup process.

Greetings
Alex

**Comment by mgoeller** [comment]
No, I dont think this is it since I tried adding the Accessory to a completely empty "House" in the Home App. If I can somehow help you to debug, let me know.

**Comment by AlexanderBabel** [comment]
Then try to delete the accessory and persistent folder and try to add it again.

Do you have multiple Nello one?

**Comment by mgoeller** [comment]
As written above I did this whenever I was readding the accessory. I have just one Nello one.

**Comment by AlexanderBabel** [comment]
Then I don't get it. 

So from the beginning... You have two Raspberry Pi's. One with a camera module V2.1 and the Nello plugin installed and working. And another one with the Nello plugin installed too and you are trying to use it with the ffmpeg mode.

But you only have one Nello one. So that means that you are trying to use the same Nello one twice. 

And if this is case it is not possible because of the naming issue. Also only one instance at the same time can receive data through the web hook.

**Comment by jcarldk** [comment]
Do you need to have a video stream to get the unlock in notification? 

I'm a little confused by you manual around doorbell and video.

My use case I have video Siedle intercom. (I havn't figured out yet how to get the video stream, I hope door bird ip upgrade with video will fix this for me).

So even though I don't have video stream to put in, I want to have an unlock option in my notification.

**Comment by AlexanderBabel** [comment]
Do you still need help?

**Comment by mgoeller** [comment]
Thank you, I forgot to look into it. I installed now the newest update, and used the doorbell as motion sensor. Then I just used a different camera plugin. As long as the motion sensor, door lock and camera are in the same room, a rich notification will be issued with Video and Unlock button. I would suggest you keep the nello plugin clean and advise people to do it like this? Instead of integrating the rpicamera and khaoscamera plugin into yours. The only thing I'm noticing and can't explain that from time to time the notifications come without a video screenshot. But only in seldom cases. Thank you for the development and help.
